### PR TITLE
Better explain the most common problems when testing emails

### PR DESCRIPTION
**Pull request by javiereguiluz**
Testing emails is simple but in the past I've suffered lots of problems related to this. In fact, a few minutes ago I fixed a long standing wrong test thanks to this comment about page redirections and collectors: https://github.com/symfony/symfony/issues/20282#event-848771951

I propose to create a new mini section to better explain the problems that you may face when testing emails.

### CODE REVIEW

File: email/testing.rst [hunk 1]
```diff
@@ -67,4 +63,16 @@ to get information about the messages sent on the previous request::
...
 
+Troubleshooting
+---------------
+
+The email collector is only available when the profiler is enabled and collects
+information, as explained in :doc:`/testing/profiling`.
+
+If the action that sends the email redirects to another page (for example when
+you send an email after a form is processed and before redirecting to another
+page) make sure that the test client doesn't follow the redirects, as explained
```
Comment (xabbuh) [inline email/testing.rst]: missing comma after the closing parenthesis

File: email/testing.rst [hunk 2]
```diff
@@ -67,4 +63,16 @@ to get information about the messages sent on the previous request::
...
+Troubleshooting
+---------------
+
+The email collector is only available when the profiler is enabled and collects
+information, as explained in :doc:`/testing/profiling`.
+
+If the action that sends the email redirects to another page (for example when
+you send an email after a form is processed and before redirecting to another
+page), make sure that the test client doesn't follow the redirects, as explained
+in :doc:`/testing`. Otherwise, the collector will lose the original email when
```
Comment (xabbuh) [inline email/testing.rst]: I think the last sentence is a bit misleading. It's not that the collector loses the mail, but that you are just looking at the "wrong" profile after being redirected, right?

**Comment by xabbuh** [comment]
👍 Great work Javier!

Status: Reviewed

**Comment by wouterj** [comment]
Are there any specific error messages/problem descriptions that lead to one of these 2 problems? I like troubleshooting sections, but it would be even greater if it's structured like:

```
## Problem: The mail collector returns `null`

The profiler must be enabled ...

## Problem: The email is not listed in the collections

Make sure you don't allow redirections ...
```

**Comment by javiereguiluz** [comment]
@wouterj I agree. I've added some subtitles. Thanks!

**Comment by wouterj** [comment]
(btw, my subtitles where just examples, I don't know if these are the true indicators of the problems).

If you've checked the cases: 👍 Otherwise, please check before merging :)

**Comment by xabbuh** [comment]
@javiereguiluz What about Wouter's last comment?

**Comment by javiereguiluz** [comment]
@xabbuh I've made some minor rewords.

**Comment by xabbuh** [comment]
Thank you @javiereguiluz.

**Pull request closed without merging**